nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/clustering/FlowSynchronizationIT.java

```diff
@@ -822,8 +847,9 @@ public void testUnnecessaryProcessorsAndConnectionsRemoved() throws NiFiClientEx
 
         final ProcessorEntity terminate = getClientUtil().createProcessor("TerminateFlowFile");
         final ConnectionEntity connection = getClientUtil().createConnection(generate, terminate, "success");
-        getClientUtil().startProcessor(terminate);
         getClientUtil().startProcessor(generate);
+        waitForMinQueueCount(connection.getId(), 1);
```

Comment:
This was a weird case where the `waitForQueueCount(0)` query happened before the generate processor fired its event.